crates/store/re_types/src/archetypes/segmentation_image_ext.rs

```diff
@@ -16,62 +17,47 @@ impl SegmentationImage {
     where
         <T as TryInto<TensorData>>::Error: std::error::Error,
     {
-        let mut data: TensorData = data
+        let tensor_data: TensorData = data
             .try_into()
             .map_err(ImageConstructionError::TensorDataConversion)?;
 
-        let non_empty_dim_inds = find_non_empty_dim_indices(&data.shape);
+        let non_empty_dim_inds = find_non_empty_dim_indices(&tensor_data.shape);
 
-        match non_empty_dim_inds.len() {
-            2 => {
-                assign_if_none(&mut data.shape[non_empty_dim_inds[0]].name, "height");
-                assign_if_none(&mut data.shape[non_empty_dim_inds[1]].name, "width");
+        if non_empty_dim_inds.len() != 2 {
+            return Err(ImageConstructionError::BadImageShape(tensor_data.shape));
+        }
+
+        let (blob, data_type) = match tensor_data.buffer {
```

Comment:
maybe now is the time to add a conversion function that is shared between `segmentation_image_ext` and `depth_image_ext`?